projects/04/mult/Mult.asm

```diff
@@ -9,4 +9,26 @@
 // This program only needs to handle arguments that satisfy
 // R0 >= 0, R1 >= 0, and R0*R1 < 32768.
 
-// Put your code here.
\ No newline at end of file
+@2 // set index 2 to 0
+M=0
+
+(LOOP)
+@0
+D=M

```

Comment:
Here is an (optional) ✨ performance optimisation challenge ✨ for you: it’s possible to make the program faster by moving these two instructions outside of the loop, i.e. above the `(LOOP)` label, so that they only execute a single time at the beginning of the program rather than multiple times (once per loop iteration). But right now if you move them above the label the program will break, because the `JEQ` depends upon `D` containing the current value of `R0`. How can we modify the loop body so that it still works without executing these instructions every time?

Reply:
[`b38d674` (#5)](https://github.com/wildmaples/nand2tetris/pull/5/commits/b38d674c526c1e8dbd58aff9a2cc87c2ddedbb4b) I moved the two lines outside the loop and made sure D register points to R@0 at the end of the loop.